2017/proposals/patni-harshit.md

```diff
@@ -0,0 +1,188 @@
+# Title
+2D color maps
+
+## Abstract
+
+All of the color mapping in Matplotlib is currently derived from
+`ScalerMappable` which as the name suggests maps scalers from `R^1 ->
+R^4` RGBA color space.  It is common to want to map a vector to
+colors, for example to control the alpha based on a second value in a
+scatter plot or to show the orientation of a field.
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Intermediate  | Python | tacaswell,story645 |
+
+## Technical Details
+
+- All the work will be done in Python
+- It will be in accordance with current implementation of 1D colormaps.
+- Extending the existing 1D normalization for 2D data : This can be done by
+a family of normalizers which go from data -> unit-disk or square in the
+complex plane
+- Creating color maps that go from the unit disk or square -> RGBA
+- Exposing these classes to user as either new API or extending the existing
+`ScalerMappable` API
+- Implementing a 2D color bar
+- Developing 2D color maps : They can be created by doing bi-linear
+interpolation between four colours. Also major focus would be on developing
+perceptually friendly colormaps so that color blind people have no difficulty in
+distinguishing colors
+
+## Schedule of Deliverables
+
+### May 1th - May 28th, **Community Bonding Period**
+
+- Before the official time period begins I will do some tasks listed under
+[MEP21](http://matplotlib.org/devel/MEP/MEP21.html): color and cm refactor.
+This will greatly help in understanding the current implementation of
+normalization and color mapping tools in Matplotlib. Some tasks that can be
+done as part of this are:
+    - Tidying up namespace
+    - Defining a "Color" class
+    - Making it easy to convert from one color type to another `hex -> RGB`, `RGB -> hex`, `HSV -> RGB` etc
+    - Improving construction of colormap by changing current dictionary approach
+    - Making it possible to interpolate between two or more color types in different modes
+    - Renaming `cm` module to something more descriptive
```

Comment:
Changed to "Analyzing the feasibility of renaming `cm` module to something more descriptive. It will be a major API change so this will be done only if its pros outweighs cons"